server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -139,6 +143,10 @@
         .appendFraction(NANO_OF_SECOND, 3, 9, true)
         .optionalEnd()
         .optionalStart()
+        .appendLiteral(",")
+        .appendFraction(NANO_OF_SECOND, 1, 9, false)
```

Comment:
the above has the digits minimum as this is a strict date? Do the same here?

Reply:
will fix